### PR TITLE
Disable scheduling/sending when email has problems

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
Adds logic to calculate if there are problems in an email that prevents it from being scheduled and displays the problems. It also makes it possible to cancel the scheduled sending of an email.  

## Screenshots
bild

## Changes
* Adds the deliveryProblems function with tests
* Updates and renames the blockHasErrors function, so it returns an array of Problems
* Displays errors in the popper under the actionbuttons if there are errors, and scheduling logic if there are none
* If the email has a publish date in the future it sets the actionbutton to say "Cancel" and the action to be to remove the publish date
* Disables editor if the email is scheduled

## Related issues
none

### CODE REVIEW

File: src/features/emails/l10n/messageIds.ts [hunk 1]
```diff
@@ -8,13 +8,31 @@ export default makeMessages('feat.emails', {
...
+    ),
+    EMPTY: m('Your email is empty. Go to the Compose tab to add some content.'),
+    NOT_TARGETED: m(
+      'Your email has no targets. Go to the Targets section in the Overview tab to create a Smart Search that defines your targets.'
+    ),
+    NO_SUBJECT: m(
+      'Your email has no subject line. Go to Settings in the Compose tab to add one.'
+    ),
+    TARGETS_NOT_LOCKED: m(
+      'The targets are not locked. Go to the Ready section un the Overview tab to do this.'
```
Comment (richardolsson) [inline src/features/emails/l10n/messageIds.ts:23]: I spotted this in the screenshot. 😊 

```suggestion
      'The targets are not locked. Go to the Ready section in the Overview tab to do this.'
```

Comment (ziggabyte) [reply src/features/emails/l10n/messageIds.ts:23]: Flag_of_the_United_Nations svg

**Review by richardolsson — CHANGES_REQUESTED** [review]
Really nice work on this! I love all the tests!

I found some stuff that I commented on below. Not all of them are critical to change, but they are some ways that I think the code can be improved. What do you think?

**Review by richardolsson — APPROVED** [review]
LGTM! 🟢

**Pull request merged**